swift-ticketing/test/swift_ticketing/booking_flow_test.clj

```diff
@@ -0,0 +1,106 @@
+(ns swift-ticketing.booking-flow-test
+  (:require [clojure.test :refer [deftest testing is use-fixtures]]
+            [swift-ticketing.fixtures :as fixtures]
+            [swift-ticketing.factory :as factory]
+            [swift-ticketing.client :as client]
+            [swift-ticketing.db.booking :as db-booking]
+            [camel-snake-kebab.extras :as cske]
+            [camel-snake-kebab.core :as csk]))
+
+(use-fixtures :each fixtures/clear-tables)
+
+(deftest general-ticket-booking-flow-test
+  (testing "Ticket booking (General)"
+    ;; setup event and tickets
+    (let [{:keys [db-spec]} fixtures/test-env
+          event-id (->> (client/create-event)
+                        :response
+                        (cske/transform-keys csk/->kebab-case-keyword)
+                        :event-id)
+          {:keys [ticket-type-id
+                  tickets]} (->> (client/create-general-tickets event-id)
+                                 :response
+                                 (cske/transform-keys csk/->kebab-case-keyword))
+          total-tickets (count tickets)
+          reserve-tickets (fn [quantity]
+                            (->> (factory/mk-reserve-general-ticket-request
+                                  quantity
+                                  ticket-type-id)
+                                 (client/reserve-ticket event-id)))
+          get-booking-id #(get % "booking_id")]
+
+      (testing "Reserving more tickets than available"
+        ;; Reserving more tickets than available should return REJECTED
+        (let [{:keys [status response]} (reserve-tickets (inc total-tickets))
+              booking-id (get-booking-id response)]
+          (is (= 201 status))
+          (Thread/sleep 2000)
```

Comment:
2 ways to test an async function:

1. Make the async part synchronous only for the tests using some config.
2. Poll from the test and the polling fn fulfils the promise.